projects/kit/src/lib/masks/number/processors/create-fullwidth-to-halfwidth-preprocessor.ts

```diff
@@ -0,0 +1,24 @@
+import {MaskitoPreprocessor} from '@maskito/core';
+
+/**
+ * Convert full width numbers like １, ２ to half width numbers 1, 2
+ *
+ * 全角の１、２ののような数字を、半角の1、2のように直すプリプロセッサ
+ */
+export function createFullWidthToHalfWidthPreprocessor(): MaskitoPreprocessor {
+    return ({elementState, data}) => {
+        const {value, selection} = elementState;
+
+        return {
+            elementState: {
+                selection,
+                value: value.replace(/[０-９]/g, s =>
+                    String.fromCharCode(s.charCodeAt(0) - 0xfee0),
+                ),
+            },
+            data: data.replace(/[０-９]/g, s =>
+                String.fromCharCode(s.charCodeAt(0) - 0xfee0),
+            ),
```

Comment:
Let create function `toHalfWidthNumber` inside [`kit/src/lib/masks/number/utils`](https://github.com/taiga-family/maskito/tree/main/projects/kit/src/lib/masks/number/utils).

```ts
export function toHalfWidthNumber(fullWidthNumber: string): string {
    return fullWidthNumber.replace(/[０-９]/g, s =>
        String.fromCharCode(s.charCodeAt(0) - 0xfee0),
    )
}
```

and use it inside this preprocessor

```suggestion
                value: toHalfWidthNumber(value),
            },
            data: toHalfWidthNumber(data),
```

Reply:
Could I also ask you to add some unit tests for this new utility `toHalfWidthNumber`-function (inside [`kit/src/lib/masks/number/utils/tests`](https://github.com/taiga-family/maskito/tree/main/projects/kit/src/lib/masks/number/utils/tests)) ?

```ts
import {stringifyNumberWithoutExp} from '../to-half-width-number';

describe('`toHalfWidthNumber` utility converts full width numbers to half width numbers', () => {
    it('０ => 0', () => {
        expect(toHalfWidthNumber('０')).toBe('0');
    });

    // [...] 
    // 2, 3, ..., 7, 8 

    it('９ => 9', () => {
        expect(toHalfWidthNumber('９')).toBe('9');
    });

    it('１２３４５６ => 123456', () => {
        expect(toHalfWidthNumber('１２３４５６')).toBe('123456');
    });

    it('１2３4５6 (full width + half width mix) => 123456', () => {
        expect(toHalfWidthNumber('１2３4５6 ')).toBe('123456');
    });
});
```

Reply:
Yes sure